engine/src/flutter/shell/platform/common/isolate_scope.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/logging.h"
+#include "third_party/dart/runtime/include/dart_api.h"
+
+#ifndef FLUTTER_SHELL_PLATFORM_COMMON_ISOLATE_SCOPE_H_
+#define FLUTTER_SHELL_PLATFORM_COMMON_ISOLATE_SCOPE_H_
+
+namespace flutter {
+
+/// This class is a thin wrapper around dart isolate. It can be used
+/// as argument to IsolateScope constructor to enter and exit the isolate.
+class Isolate {
+ public:
+  static Isolate Current();
```

Comment:
Might be worth commenting that calling this crashes if no Dart Isolate is current.